packages/xstate-wallet/src/workflows/create-and-fund-ledger.ts

```diff
@@ -0,0 +1,184 @@
+
+type WorkflowActions = {
+  assignChannelId: AssignAction<WorkflowContext, DoneInvokeEvent<string>>;
+};
+type WorkflowGuards = {
+  doesChannelIdExist: ConditionPredicate<WorkflowContext, WorkflowEvent>;
+};
+export type WorkflowEvent = AnyEventObject;
+export type WorkflowContext = {
+  initialOutcome: SimpleAllocation;
+  participants: Participant[];
+  ourIndex: number;
+  ledgerId?: string;
+};
+export type LedgerIdExists = WorkflowContext & {ledgerId: string};
+export interface WorkflowServices extends Record<string, ServiceConfig<WorkflowContext>> {
+  createObjective: (context: LedgerIdExists, event: any) => Promise<void>;
+  initializeChannel: (context: WorkflowContext, event: WorkflowEvent) => Promise<string>;
+  supportState: StateMachine<any, any, any>;
+  getPreFundState: (context: LedgerIdExists, event: WorkflowEvent) => Promise<SupportState.Init>;
+  depositing: StateMachine<any, any, any>;
+  getDepositingInfo: (context: LedgerIdExists, event: any) => Promise<Depositing.Init>;
+}
+
+export const config: StateNodeConfig<WorkflowContext, any, any> = {
+  initial: 'isChannelIdDefined',
+  on: {[ActionTypes.ErrorCustom]: {target: 'failure'}},
+  states: {
+    isChannelIdDefined: {
+      on: {
+        '': [{cond: 'doesChannelIdExist', target: 'waitForObjective'}, {target: 'initialize'}]
+      }
+    },
+    initialize: {
+      initial: 'initializeChannel',
+      states: {
+        initializeChannel: {
+          invoke: {
+            src: 'initializeChannel',
+            onDone: {target: 'createObjective', actions: []}
+          }
+        },
+        createObjective: {
+          invoke: {
+            src: 'createObjective',
+
+            onDone: [{target: 'done'}]
+          }
+        },
+        done: {type: 'final'}
+      },
+      onDone: 'supportPreFundState'
+    },
+
+    supportPreFundState: getDataAndInvoke<WorkflowContext, any>(
+      {src: 'getPreFundState'},
+      {src: 'supportState'},
+      'fundChannel'
+    ),
+
+    fundChannel: getDataAndInvoke<WorkflowContext, any>(
+      {src: 'getDepositingInfo'},
+      {src: 'depositing'},
+      'done'
+    ),
+
+    done: {type: 'final'},
+    failure: {type: 'final'}
+  }
+};
+
+const initializeChannel = (
+  store: Store
+): WorkflowServices['initializeChannel'] => async context => {
+  const stateVars: StateVariables = {
+    outcome: context.initialOutcome,
+    turnNum: bigNumberify(0),
+    isFinal: false,
+    appData: '0x0'
+  };
+  const entry = await store.createChannel(context.participants, CHALLENGE_DURATION, stateVars);
+  return entry.channelId;
+};
+
+const createObjective = (store: Store): WorkflowServices['createObjective'] => async context => {
+  const objective: Objective = {
+    type: 'FundLedger',
+    participants: context.participants,
+    data: {ledgerId: context.ledgerId}
+  };
+  return store.addObjective(objective);
+};
+const getPreFundState = (store: Store): WorkflowServices['getPreFundState'] => async context => {
```

Comment:
```suggestion
const getPreFundState = (store: Store): WorkflowServices['getPreFundState'] => async (context: LedgerIdExists) => {
```

Reply:
There's no need to type `context` it is already typed by `WorkflowServices['getPreFundState'] `